### PR TITLE
Nate/feature/LOOP-309/swift ui color palette

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-309

example of using the color palette from the environment

Also see https://github.com/tidepool-org/Loop/pull/181

### CODE REVIEW

File: LoopKitUI/Extensions/Environment+ColorPalette.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+//  LoopKitUI
+//
+//  Created by Nathaniel Hamming on 2020-07-29.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+private struct ColorPaletteKey: EnvironmentKey {
+    static var defaultValue: ColorPalette? = nil
```
Comment (mpangburn) [inline LoopKitUI/Extensions/Environment+ColorPalette.swift:12]: I think we should make this non-optional, even if the default colors are bogus, for the sake of avoiding optionality where we don't really expect it

File: LoopKitUI/Models/ColorPalette.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+//  LoopKitUI
+//
+//  Created by Nathaniel Hamming on 2020-07-29.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct ColorPalette {
+    public var theColor: Color?
```
Comment (mpangburn) [inline LoopKitUI/Models/ColorPalette.swift:12]: (and similarly I think the colors within the palette should be non-optional as well)

File: LoopKitUI/View Controllers/DismissableHostingController.swift [hunk 3]
```diff
@@ -8,7 +8,6 @@
 
 import SwiftUI
 
-
 public class DismissibleHostingController: UIHostingController<AnyView> {
```
Comment (mpangburn) [inline LoopKitUI/View Controllers/DismissableHostingController.swift:11]: I think we should rename this to just 'HostingController' to make it clear that it serves as our main bridge to SwiftUI for all of our custom environment data (which was originally just `dismiss`)

**Comment by mpangburn** [comment]
A couple notes on optionality, but the general approach LGTM!

File: LoopKitUI/View Controllers/DismissableHostingController.swift [hunk 4]
```diff
@@ -21,11 +20,12 @@ public class DismissibleHostingController: UIHostingController<AnyView> {
         rootView: Content,
         dismissalMode: DismissalMode = .modalDismiss,
         isModalInPresentation: Bool = true,
-        onDisappear: @escaping () -> Void = {}
+        onDisappear: @escaping () -> Void = {},
+        colorPalette: ColorPalette? = nil
```
Comment (mpangburn) [inline LoopKitUI/View Controllers/DismissableHostingController.swift:24]: Actually, this is a little tricky.. it means everywhere we instantiate one of these hosting controllers, the real palette will have to be accessible, which means we can't instantiate one from within LoopKit. What do you think about making a public static variable that we assign once from Loop when the app launches? I wouldn't normally advocate for global state, but I don't think we have a lot of options.

File: LoopKitUI/Models/ColorPalette.swift (hunk 2 above)
Comment (ps2) [inline LoopKitUI/Models/ColorPalette.swift:12]: I'm not sure I understand how this will be used. There is only one color here?

**Review by ps2 — COMMENTED** [review]
How is this being used? Should I be looking in a different PR to see that?

File: LoopKitUI/Models/ColorPalette.swift (hunk 2 above)
Comment (nhamming) [reply LoopKitUI/Models/ColorPalette.swift:12]: Sorry, this is an old PR. Deleting.

**Pull request closed without merging**